pylint/testutils/primer.py

```diff
@@ -0,0 +1,69 @@
+import os
+import shutil
+from pathlib import Path
+from typing import List, NamedTuple, Optional
+
+import git
+
+PRIMER_DIRECTORY_PATH: Path = Path(".pylint_primer_tests")
+
+
+class PackageToLint(NamedTuple):
+    """Represents data about a package to be tested during primer tests"""
+
+    url: str
+    """URL of the repository to clone"""
+
+    branch: str
+    """Branch of the repository to clone"""
+
+    commit: Optional[str]
+    """Commit hash to pin the repository on"""
+
+    @property
+    def clone_directory(self) -> str:
+        """Directory to clone repository into"""
+        return str(
+            PRIMER_DIRECTORY_PATH
+            / "/".join(self.url.split("/")[-2:]).replace(".git", "")
+        )
+
+    directories: str
+    """Directories within the repository to run pylint over"""
+
+    pylint_additional_args: List[str] = []
+    """Arguments to give to pylint"""
+
+    pylintrc: str = "./pylintrc"
```

Comment:
This doesn't work as we need to add the `clone_directory` to the path.

Reply:
c383b0937b4ca4f2b9c4acd35bb60d5a3b16c049